File: src/server/InvestmentApp-Server/Models/Period.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace InvestmentApp.Models;

public class Period
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    public DateTimeOffset StartDate { get; set; }

    public DateTimeOffset? EndDate { get; set; }  // null -> indefinitely

    public double DiscountRate { get; set; }

    public double RiskFreeDiscountRate { get; set; }
}

File: src/client/src/app/models/period.ts
import { DataTableRow } from ".";
import { Guid } from "guid-typescript";

export interface Period {
    id: Guid
    startDate: Date,
    endDate: Date,
    discountRate: number;
    riskFreeDiscountRate: number;
}

export interface PeriodRow extends DataTableRow {
    id: string,
    internalId: string;
    startDate: string,
    endDate: string,
    discountRate: number;
    riskFreeDiscountRate: number;
}
